File: 349. Intersection of Two Arrays/solution_3.java
//24%
//O(nlogn)
//排序+双指针
class Solution {
    public int[] intersection(int[] nums1, int[] nums2) {
        if (nums1.length == 0 || nums2.length == 0 || nums1 == null || nums2 == null)
            return new int[0];
        Arrays.sort(nums1);
        Arrays.sort(nums2);
        HashSet<Integer> set = new HashSet<>();
        int i = 0, j = 0;
        while (i < nums1.length && j < nums2.length) {
            if (nums1[i] < nums2[j])
                i++;
            else if (nums1[i] > nums2[j])
                j++;
            else {
                set.add(nums1[i]);
                i++;
                j++;
            }
        }
        int[] res = new int[set.size()];
        int k = 0;
        for (int num : set) {
            res[k++] = num;
        }
        return res;
    }
}

File: 227. Basic Calculator II/solution_3.cpp
class Solution {
public:
    int calculate(string s) {
        istringstream in("+" + s + "+");
        int res = 0, tmp = 0, num;
        char op;
        while (in >> op) {
            if (op == '+' || op == '-') {
                res += tmp;
                in >> tmp;
                tmp *= (op == '+') ? 1 : -1;
            }
            else {
                in >> num;
                if (op == '*')
                    tmp *= num;
                else if (op == '/')
                    tmp /= num;
            }
        }
        return res;
    }
};